### PR TITLE
Multitool can be used to find the area's APC

### PR DESCRIPTION
**Pull request by Runian**
## About The Pull Request
Ported from https://github.com/cmss13-devs/cmss13/pull/7020

Having at least one point in engineering lets you use a multitool in hand which helps you find the APC in the area.
image

## Why It's Good For The Game
There is this one intel computer that no one ever does in the streets of LV759 because the APC for it must be in the 5th dimension. Since I can't find it, I need to code something to help me (and everyone else) find it.

... And it also helps new engineers do their job by finding the broken APC that they need to fix for objectives.

## Changelog
:cl:
add: If you have any engineering skills, you can use a multitool will help you find the APC of the room you're in.
/:cl:

### CODE REVIEW

**Comment by AusarAtkin** [comment]
Speed merge this shit , for fellow visually imparied engis.

**Comment by iforgotmeword** [comment]
yes but make it have a cast time if you are not a eng instead

**Comment by SimplySmiley123** [comment]
> yes but make it have a cast time if you are not a eng instead
> 
> I don't want CBT (Combat Buddy Torture, WHY THE FUCK ARE YOU SO DUMB? JUST USE THE MULTITOOL BUDDY PLEASE)

Yeah, honestly, would be nice.

Either way the PR is already amazing and I want it for my visually impaired ass.

File: code/game/objects/items/multitool.dm [hunk 1]
```diff
@@ -18,3 +18,42 @@
 	tool_behaviour = TOOL_MULTITOOL
 
 	var/obj/machinery/telecomms/buffer // simple machine buffer for device linkage
+
+/obj/item/tool/multitool/attack_self(mob/user)
+	..()
```
Comment (TiviPlus) [inline code/game/objects/items/multitool.dm:23]:
```suggestion
	. = ..()
```

File: code/game/objects/items/multitool.dm [hunk 2]
```diff
@@ -18,3 +18,42 @@
...
+	final_image.plane = GAME_PLANE
+	final_image.loc = get_turf(area_apc)
+	final_image.dir = apc_appearance.dir
+	final_image.alpha = 225
+	user.client.images += final_image
+
+	TIMER_COOLDOWN_START(src, COOLDOWN_LOCATE_APC, 1.5 SECONDS)
+	addtimer(CALLBACK(src, PROC_REF(remove_apc_highlight), user.client, final_image), 1.4 SECONDS)
+
+/obj/item/tool/multitool/proc/remove_apc_highlight(client/user_client, image/highlight_image)
```
Comment (TiviPlus) [inline code/game/objects/items/multitool.dm:56]: autodoc pls

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**